### PR TITLE
Pin sphinx==1.5.1 and clean up dev_guide/documentation deps section

### LINKED ISSUE #6228: Task: add missing `packaging` dependency to documentation requirements
Task: add missing `packaging` dependency to documentation requirements in documentation at: http://bokeh.pydata.org/en/latest/docs/dev_guide/documentation.html#devguide-documentation

Also, ``packaging`` is listed twice in the meta.yaml file. We should clean that up.

### PR DESCRIPTION
**Pull request by canavandl**
* Pin to sphinx=1.5.1
* Remove duplicate `packaging` entry in meta.yaml
* Clean up dev_guide/documentation deps sections

- [x] issues: fixes #6228
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by bryevdv** [comment]
Is the CI build actually picking up bad versions of sphinx? If not I'd rather not pin. There fix for this has already been merged in Sphinx, I'd rather set a min version when when the next version of sphinx comes out.

File: conda.recipe/meta.yaml [hunk 1]
```diff
@@ -50,11 +49,11 @@ test:
   requires:
     # docs
     - pygments
-    - sphinx >=1.3.1
+    - sphinx >=1.4
```
Comment (canavandl) [inline conda.recipe/meta.yaml]: This matches the current conf.py (and 1.3.1 doesn't work)

File: conda.recipe/meta.yaml [hunk 2]
```diff
@@ -25,7 +25,6 @@ requirements:
     - pycurl
     - packaging
     - six
-    - packaging
```
Comment (bryevdv) [inline conda.recipe/meta.yaml]: Isn't this needed by the TravisCI docs build?

Comment (canavandl) [inline conda.recipe/meta.yaml]: it's a duplicate (look up two rows)

Comment (bryevdv) [inline conda.recipe/meta.yaml]: doh :)

**Pull request merged**